data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/client/PeerForwarderClient.java

```diff
@@ -90,4 +113,14 @@ private CompletableFuture<AggregatedHttpResponse> processHttpRequest(final WebCl
         }, executorService);
     }
 
+    private AggregatedHttpResponse getAggregatedHttpResponse(final CompletableFuture<AggregatedHttpResponse> aggregatedHttpResponseCompletableFuture) {
+        AggregatedHttpResponse response = null;
+        try {
+            response = aggregatedHttpResponseCompletableFuture.get();
+        } catch (InterruptedException | ExecutionException e) {
+            LOG.error("Problem with asynchronous peer forwarding", e);
+        }
+        return response;
```

Comment:
It might make more sense to construct and return a custom AggregatedHttpResponse than a null value.

Reply:
I'm not sure what would be a good Response status here. I think SERVICE_UNAVAILABLE should work for now.